platforms/jvm/testing-jvm-infrastructure/src/main/java/org/gradle/api/internal/tasks/testing/testng/TestNGTestClassProcessor.java

```diff
@@ -213,6 +213,16 @@ private void setThreadPoolFactoryClass(TestNG testNg, String threadPoolFactoryCl
         }
     }
 
+    private void setSuiteThreadPoolSize(TestNG testNg, Integer suiteThreadPoolSize) {
+        try {
+            JavaMethod.of(TestNG.class, Object.class, "setSuiteThreadPoolSize", Integer.class).invoke(testNg, suiteThreadPoolSize);
+        } catch (NoSuchMethodException e) {
+            if (suiteThreadPoolSize < 1) {
+                throw new InvalidUserDataException("The version of TestNG used does not support setting thread pool size.");
```

Comment:
❗ This message is incorrect, it should say something about how a size < 1 is invalid.

Please add a test to verify this failure behavior as well.

Reply:
I agree, initially I misunderstood how it works. I've made the changes and tested them—it looks like it's what we need. Please take a look.

Reply:
I understand your point. However, the main reason I did it this way is due to the logic of other methods presented in this class. Let's look at the [setPreserveOrder method](https://github.com/gradle/gradle/blob/00cf0574d7844203495faf52d33e974c4618b339/platforms/jvm/testing-jvm-infrastructure/src/main/java/org/gradle/api/internal/tasks/testing/testng/TestNGTestClassProcessor.java#L188), for example. It also throws an exception only if there is an attempt to set a value different from the default (false by default), which does not affect the program's execution.

I tested this in a separate project with an older version of TestNG (5.12.1) - I set setPreserveOrder to true and received an exception, but when I set setPreserveOrder to false, there were no exception. This is why I rewrote this method to reflect the behavior of existing methods.

However, I am new here, so if I misunderstood something, please let me know what exactly.

Reply:
OK, I see.  I was comparing to the `setSuiteThreadPoolSize` method, which I thought might be more relevant due to it also affecting parallelism.

I think this is fine then.  Can you please add an integration test that uses an older TestNG version and demonstrates this failure, and then I think this is ready to merge?

Something simple like `TestNGGroupByInstancesNotSupportedIntegrationTest` might be a good example to copy.

Reply:
I apologize for the delay. I have added a test that checks for the exception in the older version of TestNG. Please take a look.

Reply:
No worries!  Thank you for your contribution, and for following up.

This all looks great, I'm merging it now to get it added for Gradle 8.9.